compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -2936,6 +2936,10 @@ impl<'tcx> TypeRelation<'tcx> for SameTypeModuloInfer<'_, 'tcx> {
         self.0.tcx
     }
 
+    fn intercrate(&self) -> bool {
+        self.0.intercrate
```

Comment:
can you change that to
```suggestion
        assert_eq!(self.0.intercrate, false);
        false
```